api/src/main/java/org/apache/iceberg/Table.java

```diff
@@ -297,4 +297,25 @@ default AppendFiles newFastAppend() {
    * Returns a {@link LocationProvider} to provide locations for new data files.
    */
   LocationProvider locationProvider();
+
+  /**
+   * Returns the current refs for the table
+   *
+   * @return the current refs for the table
+   */
+  Map<String, SnapshotRef> refs();
+
+  /**
+   * Returns the snapshot referenced by the given name or null if no such reference exists.
+   *
+   * @return the snapshot which is referenced by the given name or null if no such reference exists.
+   */
+  default Snapshot snapshot(String name) {
```

Comment:
Implemented snapshot(String name) a default method on the interface because most of the table implementations would have this same logic.